lib/connectors_sdk/confluence_cloud/authorization.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'connectors_sdk/base/authorization'
+
+module ConnectorsSdk
+  module ConfluenceCloud
+    class Authorization < ConnectorsSdk::Base::Authorization
+      class << self
+        def access_token(params)
+          tokens = super
+          tokens.merge(:cloud_id => fetch_cloud_id(tokens['access_token']))
+        end
+
+        private
+
+        def authorization_url
+          'https://auth.atlassian.com/authorize'
+        end
+
+        def token_credential_uri
+          'https://auth.atlassian.com/oauth/token'
+        end
+
+        def oauth_scope
+          %w[
+            offline_access
+
+            read:confluence-content.all
+            read:confluence-content.summary
+            read:confluence-props
+            read:confluence-space.summary
+            read:confluence-user
+            readonly:content.attachment:confluence
+            search:confluence
+          ]
+        end
+
+        def additional_parameters
+          { :prompt => 'consent', :audience => 'api.atlassian.com' }
+        end
+
+        def fetch_cloud_id(access_token)
```

Comment:
Most of _Authorization_ logic is just specifying service type specific OAuth2 behavior. This _Cloud ID_ logic strikes me as the only part that's something totally arbitrary to the third party. Is there a reason we couldn't look up the cloud id at extractor runtime? It would mean an extra call or two per sync, but it would make further attempts to abstract _Authorization_ logic easier.